homeassistant/components/dynalite/__init__.py

```diff
@@ -16,47 +16,44 @@
 
 CONFIG_SCHEMA = cv.config_entry_only_config_schema(DOMAIN)
 
+type DynaliteConfigEntry = ConfigEntry[DynaliteBridge]
+
 
 async def async_setup(hass: HomeAssistant, config: ConfigType) -> bool:
     """Set up the Dynalite platform."""
-    hass.data[DOMAIN] = {}
-
     setup_services(hass)
 
     await async_register_dynalite_frontend(hass)
 
     return True
 
 
-async def async_entry_changed(hass: HomeAssistant, entry: ConfigEntry) -> None:
+async def async_entry_changed(hass: HomeAssistant, entry: DynaliteConfigEntry) -> None:
     """Reload entry since the data has changed."""
     LOGGER.debug("Reconfiguring entry %s", entry.data)
-    bridge = hass.data[DOMAIN][entry.entry_id]
+    bridge = entry.runtime_data
     bridge.reload_config(entry.data)
     LOGGER.debug("Reconfiguring entry finished %s", entry.data)
 
 
-async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
+async def async_setup_entry(hass: HomeAssistant, entry: DynaliteConfigEntry) -> bool:
     """Set up a bridge from a config entry."""
     LOGGER.debug("Setting up entry %s", entry.data)
     bridge = DynaliteBridge(hass, convert_config(entry.data))
     # need to do it before the listener
-    hass.data[DOMAIN][entry.entry_id] = bridge
+    entry.runtime_data = bridge
     entry.async_on_unload(entry.add_update_listener(async_entry_changed))
 
     if not await bridge.async_setup():
         LOGGER.error("Could not set up bridge for entry %s", entry.data)
-        hass.data[DOMAIN][entry.entry_id] = None
+        object.__delattr__(entry, "runtime_data")
```

Comment:
Why tho? We already do this in the entry unload in the config entry core internal thing somewhere

Reply:
God this sounds like a comment that will haunt me in the future

Reply:
`runtime_data` only gets cleared on "unload", not on "failed to setup"

Reply:
Uh what, that sounds like something we might have to look into on why we do that.

Otherwise I would recommend first doing setup and then setting runtime data

Reply:
See comment above, the code owner took time to add `# need to do it before the listener` so it feels risky to change it